File: Ex2/sanntidEx2.c
#include <pthread.h>
#include <stdio.h>

int i = 0; //Husk i og ikke j
pthread_mutex_t k;

void minus(){
	for (int j = 0; j< 1000000;j++){
        pthread_mutex_lock(&k);
		i--;
        pthread_mutex_unlock(&k);
		//printf("minus gives: %i\n", i);
	}
}

void plus(){
	for (int j = 0; j< 1000000;j++){
		pthread_mutex_lock(&k);
        i++;
        pthread_mutex_unlock(&k);
		//printf("plus gives: %i\n", i);
	}
}



int main(){
    pthread_mutex_init(&k, NULL);
    pthread_t mThread;
    pthread_t pThread;

    pthread_create(&pThread, NULL, plus, NULL);
    pthread_create(&mThread, NULL, minus, NULL);
    
    pthread_join(pThread, NULL);
    pthread_join(mThread, NULL);
    printf("done gives: %i\n", i);
    return 0;
}

File: Ex2/sanntidEx2.py

import threading
from threading import Thread


i = 0
l = threading.Lock()


def minus():
	global i
	for j in range(1000000):
		l.acquire()
		i-=1
		l.release()
		#print "minus gives: " , i, "\n"
	


def plus():
	global i
	for j in range(1000000):
		l.acquire()
		i+=1
		l.release()
		#print "plus gives: " , i, "\n"
	
	

def main():
    mThread = Thread(target = minus, args = (),)
    pThread = Thread(target = plus, args = (),)

    pThread.start()
    mThread.start()

    pThread.join()
    mThread.join()

    print "done gives: " , i, "\n"


main()